src/dsmr/parser.h

```diff
@@ -31,7 +31,11 @@
 #ifndef DSMR_INCLUDE_PARSER_H
 #define DSMR_INCLUDE_PARSER_H
 
+#ifdef ARDUINO_ARCH_AVR
 #include <util/crc16.h>
+#else
+#include "crc16.h"
+#endif
```

Comment:
I would rather see this #ifdef inside crc16.h, so that everyone just includes crc16.h, which in turn either includes util/crc16.h, or defines its own functions. Also, I think checking for __AVR__ might be better than using the ARDUINO_ARCH_AVR thing, since util/crc16.h is provided by avr-libc, not arduino (shouldn't matter much in practice, though).